adapters/beachfront/beachfront.go

```diff
@@ -0,0 +1,486 @@
+			}
+
+			beachfrontReq.Imp[videoImpsIndex].Bidfloor = beachfrontVideoExt.BidFloor
+			beachfrontReq.Imp[videoImpsIndex].Id = videoImpsIndex
+			beachfrontReq.Imp[videoImpsIndex].ImpId = imp.ID
+
+			if req.Device != nil {
+				beachfrontReq.Device.Geo.Ip = req.Device.IP
+				beachfrontReq.Device.Ua = req.Device.UA
+			}
+
+			beachfrontReq.AppId = beachfrontVideoExt.AppId
+
+		}
+	}
+
+	if req.User != nil {
+		if req.User.ID != "" {
+			//   Exchange-specific ID for the user. At least one of id or
+			//   buyeruid is recommended.
+			beachfrontReq.User.ID = req.User.ID
+		}
+
+		if req.User.BuyerUID != "" {
+			//   Buyer-specific ID for the user as mapped by the exchange for
+			//   the buyer. At least one of buyeruid or id is recommended.
+			beachfrontReq.User.BuyerUID = req.User.BuyerUID
+		}
+
+	}
+
+	if req.App != nil {
+		beachfrontReq.Domain = req.App.Domain
+		beachfrontReq.Site.Page = req.App.ID
+	} else {
+		beachfrontReq.Domain = strings.Split(strings.Split(req.Site.Page, "//")[1], "/")[0]
+		beachfrontReq.Site.Page = req.Site.Page
+	}
+
+	return beachfrontReq, nil
+}
+
+func (a *BeachfrontAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var bidResp openrtb.BidResponse
+	var err error
+	var bidtype openrtb_ext.BidType = openrtb_ext.BidTypeBanner
+	var isVideo bool = false
+
+	for i := range internalRequest.Imp {
+		if internalRequest.Imp[i].Video != nil {
+			isVideo = true
+			bidtype = openrtb_ext.BidTypeVideo
+			break
+		}
+	}
+
+	bidResp, err = postprocess(response, externalRequest, internalRequest.ID, isVideo)
+	if err != nil {
+		return nil, []error{fmt.Errorf("Failed to process the beachfront response\n%s", err)}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(BidCapacity)
+
+	var errs []error
+	for _, sb := range bidResp.SeatBid {
+		for i := 0; i < len(sb.Bid); i++ {
+
+			bid := sb.Bid[i]
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: bidtype,
+			})
+		}
+	}
+	return bidResponse, errs
+}
+
+func postprocess(response *adapters.ResponseData, externalRequest *adapters.RequestData, id string, isVideo bool) (openrtb.BidResponse, error) {
+	var beachfrontResp []BeachfrontResponseSlot
+	var openrtbResp openrtb.BidResponse
+	var err error
+
+	if isVideo {
+		// Regular video ad. Beachfront returns video ads in openRTB format (or close enough).
+		if err = json.Unmarshal(response.Body, &openrtbResp); err != nil {
+			return openrtbResp, err
+		}
+		return postprocessVideo(openrtbResp, externalRequest, id)
+	} else {
+		/* Beachfront currently returns banner ads in a sparse format which is just the openRTB seatbid
+		object. It needs to be wrapped up in openrtb format.
+		*/
+		if err = json.Unmarshal(response.Body, &beachfrontResp); err != nil {
+			return openrtbResp, err
+		}
+
+		openrtbResp.ID = id
+		for range beachfrontResp {
```

Comment:
It seems like this can all be simplified quite a bit... though correct me if I'm wrong.

`MakeBids` needs to return a `BidderResponse`, which is basically a `(currency, list of bids)` pair. Your banner endpoint returns a list of bids. However, you're wrapping each `Bid` into its own `SeatBid` (here & in `postprocessBanner`), and then _unwrapping_ it in `MakeBids`.

Wouldn't it be simpler, and more efficient, to just return the list of bids?

Reply:
Yes it would. Thanks.